### PR TITLE
Law: general heat equation

### PR DESCRIPTION
**Pull request by alesanter**

### CODE REVIEW

File: test/thermodynamics/heat_transfer/general_equation_in_one_dimension_test.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+@fixture(name="test_args")
+def test_args_fixture() -> Args:
+    rho = 1
+    cv = 1
+    k = 1
+    q = 0
+    T = sin(heat_equation.position) * exp(-1 * heat_equation.time)
+    t = 1
+    x = 1
```
Comment (blackyblack) [inline test/thermodynamics/heat_transfer/general_equation_in_one_dimension_test.py:21]: Try to look for some real numbers. This is way too synthetic.

File: test/thermodynamics/heat_transfer/general_equation_in_one_dimension_test.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    lhs = heat_equation.law.lhs.subs(values)
+    rhs = heat_equation.law.rhs.subs(values)
+    assert expr_equals(lhs, rhs)
+    values = {
+        heat_equation.position: test_args.x,
+        heat_equation.time: test_args.t,
+    }
+    lhs = lhs.subs(values)
+    rhs = rhs.subs(values)
+    assert expr_equals(lhs, rhs)
```
Comment (blackyblack) [inline test/thermodynamics/heat_transfer/general_equation_in_one_dimension_test.py:42]: Now I think this second check does not make much sense.

File: test/thermodynamics/heat_transfer/general_equation_in_one_dimension_test.py (hunk 1 above)
Comment (alesanter) [reply test/thermodynamics/heat_transfer/general_equation_in_one_dimension_test.py:21]: image
The problem is that for more "real" numbers the floating point arithmetic doesn't get on well with `assert_equal`.

Comment (alesanter) [reply test/thermodynamics/heat_transfer/general_equation_in_one_dimension_test.py:21]: This gets better if I represent the values as Rationals

File: test/thermodynamics/heat_transfer/general_equation_in_one_dimension_test.py (hunk 2 above)
Comment (alesanter) [reply test/thermodynamics/heat_transfer/general_equation_in_one_dimension_test.py:42]: Agreed

**Review by blackyblack — APPROVED** [review]

**Pull request merged**